src/iamserver/iamserver.cpp

```diff
@@ -242,6 +220,66 @@ aos::Error IAMServer::SubjectsChanged(const aos::Array<aos::StaticString<aos::cS
     return aos::ErrorEnum::eNone;
 }
 
+void IAMServer::OnCertChanged(const aos::iam::certhandler::CertInfo& info)
+{
+    mPublicCred = aos::common::utils::GetTLSServerCredentials(info, *mCertLoader, *mCryptoProvider);
+    mProtectedCred
+        = aos::common::utils::GetMTLSServerCredentials(info, mConfig.mCACert.c_str(), *mCertLoader, *mCryptoProvider);
+
+    // postpone restart so it didn't block ApplyCert
+    mCertChangedResult = std::async(std::launch::async, [this]() {
+        LOG_INF() << "async in";
+        sleep(1);
```

Comment:
Looks not so good, especially with sleep 1.

Reply:
I know

Reply:
Probably async grpc server itf might fix the problem with deferred actions with it's OnDone method.

https://grpc.io/docs/languages/cpp/callback/

Reply:
Will you try?

Reply:
no. Takes a lot of time to implement. Left std::async

Reply:
But is sleep 1 required?